app/charts/shared/chart-helpers.tsx

```diff
@@ -160,4 +168,43 @@ export const getLabelWithUnit = (
     : dimension.label;
 };
 
+export const checkForMissingValuesInSegments = (
+  groupedMap: InternMap<string, Array<Observation>>,
+  segments: Array<string>
+): boolean => {
+  for (const value of groupedMap.values()) {
+    if (value.length !== segments.length) {
+      return true;
+    }
+  }
+
+  return false;
+};
 
+export const useImputationNeeded = ({
+  chartConfig,
+  data,
+}: {
+  chartConfig: ChartConfig;
+  data?: Array<Observation>;
+}) => {
+  const getSegment = useCallback(
+    (d: Observation): string =>
+      chartConfig.fields.segment
+        ? (d[chartConfig.fields.segment.componentIri] as string)
+        : "segment",
+    [chartConfig.fields.segment]
+  );
+
+  if (isAreaConfig(chartConfig) && data) {
+    return checkForMissingValuesInSegments(
+      group(
+        data,
+        (d: Observation) => d[chartConfig.fields.x.componentIri] as string
+      ),
+      [...new Set(data.map((d) => getSegment(d)))]
+    );
+  }
```

Comment:
I think you should useMemo here so that we do not checkForMIssingValues everytime we pass into this function. I think the getSegment helper could be intiialized inside the useMemo, not sure it it serves that it is useful on its own.